apps/extension/src/background/keyring/keyring.ts

```diff
@@ -761,9 +761,13 @@ export class KeyRing {
 
     if (account.public.type === AccountType.ShieldedKeys) {
       const xsk = JSON.parse(sensitiveProps.text).spendingKey;
+      //TODO: verify if this is still needed
       // Append xsk to SDK wallet instance
       this.sdk.add_spending_key(xsk, account.public.id);
```

Comment:
We can get rid of this line. I just confirmed that it works without this with your other updates!